### PR TITLE
Stabilize tx test explicating chain sync before mining

### PR DESCRIPTION
**Pull request by lucafavatella**
Delivers https://www.pivotaltracker.com/story/show/163696452
Delivers https://www.pivotaltracker.com/story/show/165184028

### CODE REVIEW

**Comment by lucafavatella** [comment]
Test originally introduced in https://github.com/aeternity/aeternity/pull/1478

**Comment by lucafavatella** [comment]
Locally I get failure:
```
%%% aecore_txs_SUITE ==> missing_tx_gossip: FAILED
%%% aecore_txs_SUITE ==> 
Failure/Error: ?assertMatch(X when is_binary ( X ), rpc : call ( N2 , aec_chain , find_tx_with_location , [ TxH1 ] ))
  expected: = X when is_binary ( X )
       got: none
      line: 194
```

File: apps/aecore/test/aecore_txs_SUITE.erl [hunk 1]
```diff
@@ -177,16 +177,23 @@ missing_tx_gossip(Config) ->
...
+    none = rpc:call(N2, aec_chain, find_tx_with_location, [TxH1]),
+    none = rpc:call(N2, aec_chain, find_tx_with_location, [TxH4]),
+
+    %% Set the same mining_rate to validate target
+    %% Only needed when chain more than 18 blocks
+    ok = rpc:call(N2, application, set_env, [aecore, expected_mine_rate, aecore_suite_utils:expected_mine_rate()], 5000),
+    {ok, MinedKeyBlocks1} = aecore_suite_utils:mine_blocks_until_txs_on_chain(N1, [TxH1, TxH2, TxH3, TxH4], ?MAX_MINED_BLOCKS),
+    aecore_suite_utils:wait_for_height(N2, aec_blocks:height(lists:last(MinedKeyBlocks1))),
+    ?assertMatch(X when is_binary(X), rpc:call(N2, aec_chain, find_tx_with_location, [TxH1])),
+    ?assertMatch(X when is_binary(X), rpc:call(N2, aec_chain, find_tx_with_location, [TxH4])),
```
Comment (zp-sd) [inline apps/aecore/test/aecore_txs_SUITE.erl]: You are trying to find a location of a transaction encoded by by `aeser_api_encoder:encode/2`. I think this should rather be:

```
    {ok, Tx1Hash} = aeser_api_encoder:safe_decode(tx_hash, TxH1),
    {ok, Tx4Hash} = aeser_api_encoder:safe_decode(tx_hash, TxH4),
    {_BlockHash1, _} = rpc:call(N1, aec_chain, find_tx_with_location, [Tx1Hash]),
    {_BlockHash2, _} = rpc:call(N1, aec_chain, find_tx_with_location, [Tx4Hash]),
```

Comment (zp-sd) [reply apps/aecore/test/aecore_txs_SUITE.erl]: And additionally `assertMatch` on `_BlockHashX` being a binary...

**Comment by zp-sd** [comment]
I added comments... Though, I don't see how this PR fixes intermittent failures from the PT ticket. Also, the PT ticket you are trying to fix looks similar to https://www.pivotaltracker.com/n/projects/2124891/stories/165184028? 🤔

File: apps/aecore/test/aecore_txs_SUITE.erl (hunk 1 above)
Comment (lucafavatella) [reply apps/aecore/test/aecore_txs_SUITE.erl]: Applied in f097666 - thanks.

**Comment by lucafavatella** [comment]
> I don't see how this PR fixes intermittent failures from the PT ticket.

@zp-sd Notice `ok = rpc:call(N2, application, set_env, [aecore, expected_mine_rate, aecore_suite_utils:expected_mine_rate()], 5000)` combined to `aecore_suite_utils:wait_for_height(N2, aec_blocks:height(lists:last(MinedKeyBlocks1)))`.

My hypothesis is that node 2 was sometimes mining faster than syncing the chain from node 1, hence node 2 was always excluding transaction 5 for the nonce. The wait for height is meant to solve this. Also, this wait would fail because of different expected mine rate, that affects target of blocks after 18.

**Comment by lucafavatella** [comment]
Failure of one job in CI [tracked](https://www.pivotaltracker.com/story/show/165354296).

**Pull request merged**